src/main/kotlin/app/batch/PlainFileWriter.kt

```diff
@@ -1,28 +1,46 @@
 package app.batch
 
+import app.domain.DataKeyResult
+import app.domain.EncryptionResult
+import app.services.CipherService
+import app.services.KeyService
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.batch.item.ItemWriter
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Profile
 import org.springframework.stereotype.Component
 import java.io.*
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
 
 @Component
 @Profile("outputToFile")
-class FileSystemWriter : ItemWriter<String> {
+class FileSystemWriter(private val keyService: KeyService,
+                       private val cipherService: CipherService) : Writer<String>(keyService, cipherService) {
+    override fun outputPath(number: Int): Path {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+
+    @Value("\${file.output}")
+    private lateinit var outputFile: String
 
-    override fun write(items: MutableList<out String>) {
+    override fun writeData(encryptionResult: EncryptionResult, dataKeyResult: DataKeyResult) {
         logger.info("Appending to '$outputFile'.")
         val fw = FileWriter(outputFile, true)
-        items.forEach {
-            fw.write("$it\n")
-        }
+        // items.forEach {
```

Comment:
Please remove the comments

Reply:
Thanks Mark , i have incorporated all  review comments.